haas/model.py

```diff
@@ -40,17 +42,30 @@
 
 db = SQLAlchemy(app)
 
+# Sets up variant type so that postgresql can use BIGINT primary keys
+# while sqlite uses Integer primary keys.
+BigIntegerType = BigInteger().with_variant(
+        sqlite.INTEGER(), 'sqlite')
+
 
 def init_db(uri=None):
     """Start up the DB connection.
 
-    `uri` is the uri to use for the databse. If it is None, the uri from the
+    `uri` is the uri to use for the database. If it is None, the uri from the
     config file will be used.
     """
     if uri is None:
         uri = cfg.get('database', 'uri')
     app.config.update(SQLALCHEMY_DATABASE_URI=uri)
 
+    global BigIntegerType
+
+    # Decide if primary keys should be BIGINTs or Integers
+    if 'postgresql' in uri:
```

Comment:
Excellent point!

I'm still wondering why the definition of `BigIntegerType` on line 47, with its *with_variant()* exception for sqlite isn't sufficient? Why do we need to create a *with_variant()* version for postgres?